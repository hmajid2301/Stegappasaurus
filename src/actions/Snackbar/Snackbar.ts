import Snackbar from 'react-native-snackbar';

import {primary, pureWhite} from '~/constants/colors';

interface Show {
  text: string;
  buttonText?: string;
  onButtonPress?: () => void;
  duration?: number;
}

export default class CustomSnackbar {
  public static show({
    onButtonPress = () => null,
    buttonText = 'Okay',
    duration = 5000,
    text,
  }: Show) {
    Snackbar.show({
      action: {
        color: primary,
        onPress: onButtonPress,
        title: buttonText,
      },
      color: pureWhite,
      duration,
      title: text,
    });
  }
}
