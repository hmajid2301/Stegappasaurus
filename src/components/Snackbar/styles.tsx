import { StyleSheet } from "react-native";

import { colors, fonts } from "~/common/styles";

const styles = StyleSheet.create({
  buttonText: { color: colors.primary },
  snackbar: { backgroundColor: "#323232", borderRadius: 3, margin: 5 },
  text: { fontSize: 14, fontFamily: fonts.bodyLight, paddingHorizontal: 10 }
});

export default styles;
