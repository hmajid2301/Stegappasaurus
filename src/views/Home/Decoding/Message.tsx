import * as React from "react";
import { NavigationScreenProp } from "react-navigation";

import ImageMessage from "~/components/ImageMessage";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
}

interface IState {
  isDecoded: boolean;
  message: string;
  photo: string;
}

class Progress extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");
    const message = navigation.getParam("message", "NO-ID");

    this.state = {
      isDecoded: false,
      message,
      photo: uri
    };
  }

  public render() {
    return (
      <ImageMessage
        editable={false}
        message={this.state.message}
        photo={this.state.photo}
      />
    );
  }
}

export default Progress;
