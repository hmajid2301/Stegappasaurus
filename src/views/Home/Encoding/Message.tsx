import { Toast } from "native-base";
import React, { Component } from "react";
import { Image } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import ImageMessage from "~/components/ImageMessage";
import Steganography from "~/services/steganography";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
}

interface IState {
  photo: string;
}

export default class Message extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");

    this.state = {
      photo: uri
    };
  }

  public render() {
    return <ImageMessage action={this.onSubmit} photo={this.state.photo} />;
  }

  private onSubmit = (message: string) => {
    let bitsCanEncode = 0;
    Image.getSize(
      this.state.photo,
      (width, height) => {
        bitsCanEncode = (width * height * 3) / 4;
      },
      () => null
    );

    const binaryMessage = new Steganography(
      "LSB-PNG",
      []
    ).convertMessageToBinary(message);
    const messageLength = binaryMessage.join("").length;

    if (messageLength > bitsCanEncode) {
      Toast.show({
        buttonText: "Okay",
        duration: 5000,
        text: "Message too large to encode in image.",
        type: "danger"
      });
    } else {
      this.props.navigation.navigate("Progress", {
        message,
        uri: this.state.photo
      });
    }
  };
}
