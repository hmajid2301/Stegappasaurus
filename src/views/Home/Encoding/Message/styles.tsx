import { Dimensions, StyleSheet } from "react-native";

import { colors, fonts } from "~/constants";

const SCREEN_HEIGHT = Dimensions.get("screen").height;
const SCREEN_WIDTH = Dimensions.get("screen").width;

const styles = StyleSheet.create({
  container: {
    alignItems: "stretch",
    backgroundColor: "white",
    height: SCREEN_HEIGHT * 0.55,
    justifyContent: "center",
    marginHorizontal: SCREEN_WIDTH * 0.025,
    marginVertical: SCREEN_HEIGHT * 0.25,
    paddingHorizontal: 20,
    paddingVertical: 20
  },

  modalHeader: {
    fontFamily: fonts.header,
    fontSize: 25,
    marginBottom: 20,
    textAlign: "center"
  },

  sliderContainer: {
    alignItems: "stretch",
    flex: 1,
    justifyContent: "center",
    marginBottom: 20
  },

  modalText: {
    fontFamily: fonts.bodyLight
  },

  passwordContainer: {
    flex: 1
  },

  passwordTextInput: {
    borderBottomColor: "grey",
    borderBottomWidth: 1
  },

  buttonContainer: {
    flex: 1,
    marginTop: 20
  }
});

export default styles;
