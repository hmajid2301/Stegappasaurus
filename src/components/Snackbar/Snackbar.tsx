import Snackbar from "react-native-snackbar";

import { colors } from "~/constants";

interface IShow {
  text: string;
  buttonText?: string;
  onButtonPress?: () => void;
  duration?: number;
}

export default class CustomSnackbar {
  public static show = ({
    onButtonPress = () => null,
    buttonText = "Okay",
    duration = 5000,
    text
  }: IShow) => {
    Snackbar.show({
      action: {
        color: colors.primary,
        onPress: onButtonPress,
        title: buttonText
      },
      duration,
      title: text
    });
  };
}
