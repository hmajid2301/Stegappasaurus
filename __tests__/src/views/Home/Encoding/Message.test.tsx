import { mount, shallow } from "enzyme";
import React from "react";

import Message from "~/views/Home/Encoding/Message";
import Snackbar from "~/components/Snackbar";

describe("Encoding Message: Match Snapshots", () => {
  test("1", () => {
    const navigate = {
      navigate: jest.fn(),
      getParam: jest.fn()
    };
    const component = shallow(<Message navigation={navigate as any} />);
    expect(component).toMatchSnapshot();
  });
});

describe("Encoding Message: Function", () => {
  let instance: React.Component<{}, {}, any>;
  let addSpy = jest.fn();

  beforeAll(() => {
    const navigate = {
      navigate: jest.fn(),
      getParam: jest.fn(),
      addListener: addSpy
    };
    const component = mount(<Message navigation={navigate as any} />);
    instance = component.instance();
  });

  test("Empty Message", () => {
    const spy = jest.spyOn(Snackbar, "show");
    (instance as any).onSubmit("");
    expect(spy).toHaveBeenCalled();
  });

  test("Not Empty", () => {
    (instance as any).onSubmit("Test!");
    expect(addSpy).toHaveBeenCalled();
  });
});
