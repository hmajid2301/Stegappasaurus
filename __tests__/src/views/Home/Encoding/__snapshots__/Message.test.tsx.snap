// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encoding Message: Match Snapshots 1 1`] = `
<View>
  <ImageMessage
    action={[Function]}
    editable={true}
    message=""
    navigation={
      Object {
        "getParam": [MockFunction] {
          "calls": Array [
            Array [
              "uri",
              "NO-ID",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "navigate": [MockFunction],
      }
    }
  />
</View>
`;
